main.go

```diff
@@ -207,6 +207,28 @@ func extractFromFile(w io.Writer, args string) error {
 	return nil
 }
 
+// replaced by testing functions.
+var (
+	readFile = ioutil.ReadFile
+	httpGet  = http.Get
+)
+
+func readContents(path string) ([]byte, error) {
+	if !strings.HasPrefix(path, "http://") && !strings.HasPrefix(path, "https://") {
+		return readFile(path)
+	}
+
+	res, err := httpGet(path)
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("status %s", res.Status)
+	}
+	defer res.Body.Close()
```

Comment:
I think this `defer res.Body.Close()` should be moved 3 lines up, before `if res.StatusCode != http.StatusOK {` check. Otherwise the response body will not be closed if response status is not 200.